crates/iota-metrics/src/hardware_metrics.rs

```diff
@@ -0,0 +1,280 @@
+// Copyright (c) 2025 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::path::Path;
+
+use prometheus::{
+    IntGauge, Opts,
+    core::{Collector, Desc},
+    proto::{LabelPair, Metric, MetricFamily},
+};
+use sysinfo::{CpuRefreshKind, Disks, MemoryRefreshKind, RefreshKind, System};
+
+use crate::RegistryService;
+
+pub fn register_hardware_metrics(
+    registry_service: &mut RegistryService,
+) -> Result<(), HardwareMetricsErr> {
+    registry_service
+        .default_registry
+        .register(Box::new(HardwareSpecs::new()?))
+        .map_err(HardwareMetricsErr::ErrRegisterHardwareMetrics)
+}
+
+pub struct HardwareSpecs {
+    pub cpu_model: String,
+    pub cpu_vendor_id: String,
+    pub cpu_core_count: Option<usize>,
+    pub memory_collector: IntGauge,
+    pub disk_collector: IntGauge,
+    pub is_docker: bool,
+}
+impl HardwareSpecs {
+    pub fn new() -> Result<Self, HardwareMetricsErr> {
+        let mut system = System::new_with_specifics(
+            RefreshKind::nothing()
+                .with_cpu(CpuRefreshKind::nothing())
+                .with_memory(MemoryRefreshKind::nothing().with_ram()),
+        );
+        system.refresh_all();
+
+        let cpu_vendor_id: &str = {
+            let vendor_id = system
+                .cpus()
+                .first()
+                .map_or("cpu_vendor_id_unavailable", |cpu| cpu.vendor_id());
+            match vendor_id {
+                "" => "cpu_vendor_id_unavailable",
+                _ => vendor_id,
+            }
+        };
+
+        let cpu_model: &str = {
+            let brand = system
+                .cpus()
+                .first()
+                .map_or("cpu_model_unavailable", |cpu| cpu.brand());
+            match brand {
+                "" => "cpu_model_unavailable",
+                _ => brand,
+            }
+        };
+        let cpu_core_count = system.physical_core_count();
+
+        let mem_total_bytes = system.total_memory();
+        let memory_collector = IntGauge::with_opts(
+            Opts::new(
+                "memory_specs",
+                "Memory specs (constants: total amount, ...)",
+            )
+            .const_label("memory_total_bytes", mem_total_bytes.to_string())
+            .const_label("memory_total_human", human_fmt_bytes(mem_total_bytes)),
+        )
+        .map_err(HardwareMetricsErr::ErrCreateMetric)?;
+        memory_collector.set(mem_total_bytes as i64);
+
+        // We're only interested in the largest disk
```

Comment:
I wonder, could we actually figure out the mount point of the database folder, and search the correct disc via the `mount_point` func?

Reply:
Or actually we could export all of them and set disk name as a label. Sometimes system has problems if a system partition is running out of storage.

Reply:
I've done both, we now have metrics for available space for each disk, as well as extra disk info for the DB one so we can correlate in grafana